dali/python/nvidia/dali/_utils/eager_util.py

```diff
@@ -0,0 +1,129 @@
+        else:
+            data = _tensors.TensorListCPU(data, layout or "")
+
+    return data
+
+
+class _Classification:
+    """Classification of data's device and if it is a batch.
+
+    Based on data type determines if data should be treated as a batch and with which device.
+    If the type can be recognized as a batch without being falsely categorized as such, it is.
+    This includes lists of supported tensor-like objects e.g. numpy arrays (the only list not
+    treated as a batch is a list of objects of primitive types), :class:`DataNodeDebug` and
+    TensorLists.
+
+    Args:
+        data: Data to be classified.
+        type_name (str): Representation of argument type (input or keyward).
+        arg_batch_size (int): Size of batch to which constant value should be scaled up. Only
+            applicable for argument inputs that are array types (e.g. numpy array). If -1 does not
+            modify the data.
+    """
+
+    def __init__(self, data, type_name, arg_batch_size=-1):
+        self.is_batch, self.device, self.data = self._classify_data(data, type_name, arg_batch_size)
+
+    @staticmethod
+    def _classify_data(data, type_name, arg_batch_size):
+        from nvidia.dali._debug_mode import DataNodeDebug
+        """Returns tuple (is_batch, device, unpacked data). """
+
+        def is_primitive_type(x):
+            return isinstance(x, (int, float, bool, str))
+
+        def classify_array_type(data):
+            if _types._is_numpy_array(data):
+                device = 'cpu'
+            elif _types._is_torch_tensor(data):
+                device = 'gpu' if data.is_cuda else 'cpu'
+                
+                # Data needs to be on CPU for batch argument inputs.
+                if device == 'gpu' and arg_batch_size > 0:
+                    data = data.cpu().numpy()
+            elif _types._is_mxnet_array(data):
+                import mxnet as mx
+
+                device = 'gpu' if 'gpu' in str(data.context) else 'cpu'
+
+                # For batch copy to CPU.
+                if device == 'gpu' and arg_batch_size > 0:
+                    data = data.copyto(mx.cpu())
+            else:
+                raise RuntimeError(f"Unsupported array type '{type(data)}'.")
+            
+            data = _types._preprocess_constant_array_type(data)[0]
+            if arg_batch_size > 0:
+                data = _tensors.TensorListCPU([_tensors.TensorCPU(data)] * arg_batch_size)
+                return True, 'cpu', data
+
+            return False, device, data
+
+        if isinstance(data, list):
+            if len(data) == 0 or any([is_primitive_type(d) for d in data]):
+                return False, 'cpu', data
+
+            is_batch_list = []
+            device_list = []
+            data_list = []
+
+            for d in data:
+                is_batch, device, val = _Classification._classify_data(
+                    d, type_name, arg_batch_size=False)
```

Comment:
`arg_batch_size=False` but in `_classify_data` it's compared as `arg_batch_size > 0`

Reply:
done